core/ecschema-metadata/src/Context.ts

```diff
@@ -9,10 +9,16 @@ import { MutableSchema, Schema } from "./Metadata/Schema";
 import { SchemaItem } from "./Metadata/SchemaItem";
 import { SchemaItemKey, SchemaKey } from "./SchemaKey";
 
+interface SchemaInfo {
+  schema: Schema;
+  startLoading?: () => Promise<Schema>;
+  loadSchema?: Promise<Schema>;

```

Comment:
Having this function and the promise bothers me ... I'm not sure exactly what we should do but this seems clunky